Source/WebCore/platform/image-decoders/png/PNGImageDecoder.cpp

```diff
@@ -66,8 +66,11 @@ const double cMaxGamma = 21474.83;
 const double cDefaultGamma = 2.2;
 const double cInverseGamma = 0.45455;
 
-// Protect against large PNGs. See Mozilla's bug #251381 for more info.
-const unsigned long cMaxPNGSize = 1000000UL;
+// Protect against APNGs with huge amounts of frames, and PNGs with large amounts
+// of pixel data un general. See bug #302220 and Mozilla's bug #251381 for more info.
+static constexpr unsigned long cMaxFrameCount = (1 << 18) - 1;
+static constexpr unsigned long cMaxPNGSize = 1000000UL;
```

Comment:
Let's not mess around with silly types like "long":

```suggestion
static constexpr uint32_t cMaxFrameCount = (1 << 18) - 1;
static constexpr uint32_t cMaxPNGSize = 1000000;
```